subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/util/ResponseHelper.java

```diff
@@ -183,30 +183,34 @@ public ResponseBuilder setInfoHeaders( final ResponseBuilder builder, final Tran
             for ( final Map.Entry<String, List<String>> headerSet : exchangeMetadata.getResponseHeaders().entrySet() )
             {
                 final String key = headerSet.getKey();
-                if ( ApplicationHeader.content_type.upperKey().equals( key ) )
+                final List<String> val = headerSet.getValue();
+                if ( val != null && !val.isEmpty() )
                 {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.content_type.upperKey() );
-                    conTypeSet = true;
-                }
-                else if ( ApplicationHeader.last_modified.upperKey().equals( key ) )
-                {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.last_modified.upperKey() );
-                    lastModSet = true;
-                }
-                else if ( ApplicationHeader.content_length.upperKey().equals( key ) )
-                {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.content_length.upperKey() );
-                    lenSet = true;
-                    if ( !includeContentLength )
+                    if ( ApplicationHeader.content_type.upperKey().equals( key ) )
                     {
-                        continue;
+                        logger.debug( "Marking {} as already set.", ApplicationHeader.content_type.upperKey() );
+                        conTypeSet = true;
+                    }
+                    else if ( ApplicationHeader.last_modified.upperKey().equals( key ) )
+                    {
+                        logger.debug( "Marking {} as already set.", ApplicationHeader.last_modified.upperKey() );
+                        lastModSet = true;
+                    }
+                    else if ( ApplicationHeader.content_length.upperKey().equals( key ) )
+                    {
+                        logger.debug( "Marking {} as already set.", ApplicationHeader.content_length.upperKey() );
+                        lenSet = true;
+                        if ( !includeContentLength )
+                        {
+                            continue;
+                        }
                     }
-                }
 
-                for ( final String value : headerSet.getValue() )
-                {
-                    logger.debug( "Setting header: '{}'= '{}'", key, value );
-                    builder.header( key, value );
+                    for ( final String value : headerSet.getValue() )
+                    {
+                        logger.debug( "Setting header: '{}'= '{}'", key, value );
+                        builder.header( key, value );
```

Comment:
I guess here we need this?
if isNotBlank(value) { builder.header(key, value); }
It seems in this case the if ( val != null && !val.isEmpty() ) is not enough. The val array is not empty, but the element in it is empty string.

Reply:
Good idea to cover also whitespace strings if that's what you meant by "isNotBlank".

Reply:
Ah, sorry, just noticed val is a List. You are right, it should not only check if the list is not empty, but also if the list elemts are not blank strings. So it should not be checked only here, but in the added if above to avoid getting e.g. conTypeSet set to true and not put any value into the builder. It would result in skipping the adding of the default value below.

Reply:
Um... seems so. I'll change the code. Thanks for checking.